pages/implementers/interfaces.md

```diff
@@ -0,0 +1,245 @@
+This prevalence is fantastic for user choice \-- as someone wishing to use JSON Schema you can be almost certain you\'ll find an implementation suitable for your target environment.
+
+But when it comes to community support, it can be challenging to know how to perform various JSON Schema operations in a particular library or implementation, as each may have slightly differing (or more than slightly differing) APIs.
+This can become particularly challenging when considering behavior not necessarily proscribed or required by the specification itself, but which is nonetheless common, useful or even required behavior *in practice* in order to use JSON Schema.
+This page serves to document a number of these *common operations*, documenting the interfaces offered by many such implementations and inspired in part by the interfaces required for Bowtie\'s own operation.
+
+For each, we first name and describe the operation using terminology from the specification where available, or otherwise using terminology that is aimed to be succinct but precise.
+The intention of this page is partially to be a reference of important JSON Schema operations as well as a way of tailoring help for a particular language and implementation.
+We therefore include examples of each interface or piece of functionality across implementations, when known.
+The ordering of sections on this page is not necessarily indicative of their importance relative to each other.
+Omission of functionality (i.e. if you do not find the way to do something below with a given implementation) is *not* necessarily a sign that the implementation does not support the functionality, though it might mean it could not be easily found by the page authors.
+Contributions and corrections from implementers are particularly welcome, as are documentation links!
+
+For purposes of making some sections of this page more precise, we assume the existence of a set of *abstract types* all of which may be referenced below, and which will have specific concrete types within a given programming language or implementation.
+In addition to placeholder types like `String`, `Number`, `Boolean`, `Mapping`, `Callable` and the like, the JSON Schema-related types include:
+
+::: glossary
+`Schema`
+
+> The type of JSON Schemas, which may also differ across dialects of JSON Schema.
+> For common or modern versions of JSON Schema this type is essentially `Map<String, Any> | Boolean`.
+
+`Instance`
+
+> The type of JSON instances.
+> This type should essentially be `Any` or a type capable of representing all JSON values, given JSON Schema\'s applicability to any representable JSON.
+
+`Dialect`
+
+> The type of JSON dialects or dialect identifiers.
+> This type may simply be `String` if the dialect is represented within the implementation by its URI or by some short name.
+
+`EvaluationContext`
```

Comment:
The description sounds like options/configuration to me.

For me, the "context" effectively manages a single evaluation.  It carries all of the data that a schema needs throughout the process and then gets thrown away.

Options include all of the things below (whether `format` validates, custom registries, etc).  In my implementation, the context _carries_ a copy of the options, but they are separate concepts.  My implementation only exposes the context 
to support users creating extension keywords.

Reply:
Is there value in separating the two concepts? It seems simpler to flatten them and consider all of what you're calling options to be part of one big context which is indeed per-evaluation, otherwise the distinction needs to be explained at even more length -- whether you can do it on each evaluation or just once ahead of time is simply a programmer convenience is it not?

Reply:
To be clear -- this is trying to be what you call "options", and not stateful for the evaluation -- i.e. it is not the place annotations are being built up inside.

(What word used seems malleable)

But I haven't here represented what you call "context" in your implementation at all I think, so I guess I'm asking whether you see value in adding a new concept here.

Reply:
I don't necessarily see value in including what I call a context.  That's likely an implementation detail and (aside from the advanced use case of custom keywords) is completely hidden from the user.

However, I think "options" or "configuration" are likely better words for this idea.